add new macros, to avoid using language constructs in translations.

- $PHORUM['file_space_quota'] -> %file_space_quota%
- str_replace(";", ", ", $PHORUM['file_types']) -> %file_types%
- $PHORUM['max_file_size'] -> %max_file_size%

detailed list:
- $PHORUM[title] -> %phorum_title%
        "EmailVerifyBody"       =>
        "LostPassEmailSubject"  =>
        "LostPassEmailBody1"    =>
        "LostPassEmailBody2"    =>
        "PMNotifyMessage"       =>
        "PMNotifySubject"       =>
        "RegApprovedEmailBody"  =>
        "VerifyRegEmailBody1"   =>
        "VerifyRegEmailBody2"   =>
- phorum_get_url(PHORUM_LOGIN_URL) -> %login_url%
        "LostPassEmailBody2"    =>
        "RegApprovedEmailBody"  =>
        "VerifyRegEmailBody2"   =>

--- phorum-5.2.15a/include/api/file.php	2010-10-22 00:35:13.428590924 +0300
+++ phorum-5.2.15a/include/api/file.php	2010-10-22 00:46:32.460815378 +0300
@@ -201,9 +201,10 @@
         // Check if the file doesn't exceed the maximum allowed file size.
         if (isset($file["filesize"]) && $PHORUM["max_file_size"] > 0 &&
             $file["filesize"] > $PHORUM["max_file_size"]*1024) {
+            $msg = str_replace("%max_file_size%", $PHORUM['max_file_size'], $PHORUM["DATA"]["LANG"]["FileTooLarge"]);
             return phorum_api_error_set(
                 PHORUM_ERRNO_NOACCESS,
-                $PHORUM["DATA"]["LANG"]["FileTooLarge"]
+                $msg
             );
         }
 
@@ -212,9 +213,10 @@
             $sz = phorum_db_get_user_filesize_total($PHORUM["user"]["user_id"]);
             $sz += $file["filesize"];
             if ($sz > $PHORUM["file_space_quota"]*1024) {
+                $msg = str_replace("%file_space_quota%", $PHORUM['file_space_quota'], $PHORUM["DATA"]["LANG"]["FileOverQuota"]);
                 return phorum_api_error_set(
                     PHORUM_ERRNO_NOACCESS,
-                    $PHORUM["DATA"]["LANG"]["FileOverQuota"]
+                    $msg
                 );
             }
         }
@@ -236,9 +238,10 @@
 
             // Check if the extension for the file is an allowed extension.
             if (!in_array($ext, $allowed_exts)) {
+                $msg = str_replace('%file_types%', str_replace(";", ", ", $PHORUM['file_types']), $PHORUM["DATA"]["LANG"]["FileWrongType"]);
                 return phorum_api_error_set(
                     PHORUM_ERRNO_NOACCESS,
-                    $PHORUM["DATA"]["LANG"]["FileWrongType"]
+                    $msg
                 );
             }
         }
--- phorum-5.2.15a/include/controlcenter/files.php	2010-10-22 00:35:13.265257620 +0300
+++ phorum-5.2.15a/include/controlcenter/files.php	2010-10-22 00:42:47.237295406 +0300
@@ -101,7 +101,11 @@
 }
 
 if($PHORUM["file_types"]){
-    $PHORUM["DATA"]["FILE_TYPE_LIMIT"]=$PHORUM["DATA"]["LANG"]["FileTypeLimits"];
+    $PHORUM["DATA"]["FILE_TYPE_LIMIT"]= str_replace(
+      '%file_types%', 
+      str_replace(";", ", ", $PHORUM['file_types']),
+      $PHORUM["DATA"]["LANG"]["FileTypeLimits"]
+    );
 }
 
 if($PHORUM["file_space_quota"]){
--- phorum-5.2.15a/include/email_functions.php~	2010-11-04 12:58:50.000000000 +0200
+++ phorum-5.2.15a/include/email_functions.php	2010-11-04 14:22:21.362512810 +0200
@@ -300,8 +300,8 @@
             include(PHORUM_INCLUDES_DIR."/lang/{$PHORUM['language']}.php");
         }
 
-        $mail_data["mailmessage"] = $PHORUM["DATA"]["LANG"]['PMNotifyMessage'];
-        $mail_data["mailsubject"] = $PHORUM["DATA"]["LANG"]['PMNotifySubject'];
+        $mail_data["mailmessage"] = str_replace("%phorum_title%",  $PHORUM['title'], $PHORUM["DATA"]["LANG"]['PMNotifyMessage']);
+        $mail_data["mailsubject"] = str_replace("%phorum_title%",  $PHORUM['title'], $PHORUM["DATA"]["LANG"]['PMNotifySubject']);
 
         $addresses = array();
         foreach ($users as $user) {
--- phorum-5.2.15a/include/controlcenter/users.php~	2010-11-04 12:58:48.000000000 +0200
+++ phorum-5.2.15a/include/controlcenter/users.php	2010-11-04 14:53:22.786298453 +0200
@@ -49,8 +49,8 @@
             	
                 $userdata["active"]=PHORUM_USER_ACTIVE;
                 // send reg approved message
-                $maildata["mailsubject"]=$PHORUM["DATA"]["LANG"]["RegApprovedSubject"];
-                $maildata["mailmessage"]=wordwrap($PHORUM["DATA"]["LANG"]["RegApprovedEmailBody"], 72);
+                $maildata["mailsubject"]=str_replace("%phorum_title%",  $PHORUM['title'], $PHORUM["DATA"]["LANG"]["RegApprovedSubject"]);
+                $maildata["mailmessage"]=wordwrap(str_replace("%phorum_title%",  $PHORUM['title'], $PHORUM["DATA"]["LANG"]["RegApprovedEmailBody"]), 72);
                 phorum_email_user(array($user["email"]), $maildata);
                 
             }
--- phorum-5.2.15a/include/controlcenter/email.php~	2010-11-03 17:22:38.000000000 +0200
+++ phorum-5.2.15a/include/controlcenter/email.php	2010-11-03 18:23:56.130121394 +0200
@@ -64,8 +64,9 @@
             $conf_code= mt_rand ( 1000000, 9999999);
             $_POST['email_temp']=$_POST['email']."|".$conf_code;
             // ... send email ... //
+            $email_message = str_replace("%phorum_title%",  $PHORUM['title'], $PHORUM['DATA']['LANG']['EmailVerifyBody']);
             $maildata=array(
-                'mailmessage'   => wordwrap($PHORUM['DATA']['LANG']['EmailVerifyBody'], 72),
+                'mailmessage'   => wordwrap($email_message, 72),
                 'mailsubject'   => $PHORUM['DATA']['LANG']['EmailVerifySubject'],
                 'uname'         => $PHORUM['DATA']['PROFILE']['username'],
                 'newmail'       => $_POST['email'],
--- phorum-5.2.15a/htdocs/login.php	2010-11-03 18:26:22.469689876 +0200
+++ phorum-5.2.15a/htdocs/login.php	2010-11-03 18:34:26.970474420 +0200
@@ -169,11 +169,16 @@
 
                 // Mail the new confirmation code to the user.
                 $verify_url = phorum_get_url(PHORUM_REGISTER_URL, "approve=".$tmp_user["password_temp"]."$uid");
+                $replace = array(
+                    "%phorum_title%" => $PHORUM['title'],
+                    "%login_url%" => phorum_get_url(PHORUM_LOGIN_URL),
+                );
+
                 $maildata["mailsubject"] = $PHORUM["DATA"]["LANG"]["VerifyRegEmailSubject"];
                 $maildata["mailmessage"] =
-                   wordwrap($PHORUM["DATA"]["LANG"]["VerifyRegEmailBody1"],72).
+                   wordwrap(str_replace(array_keys($replace), array_values($replace), $PHORUM["DATA"]["LANG"]["VerifyRegEmailBody1"]),72).
                    "\n\n$verify_url\n\n".
-                   wordwrap($PHORUM["DATA"]["LANG"]["VerifyRegEmailBody2"],72);
+                   wordwrap(str_replace(array_keys($replace), array_values($replace), $PHORUM["DATA"]["LANG"]["VerifyRegEmailBody2"]),72);
                 phorum_email_user(array($user["email"]), $maildata);
 
                 $okmsg = $PHORUM["DATA"]["LANG"]["RegVerifyEmail"];
@@ -199,14 +204,19 @@
                 // Mail the new password.
                 $user = phorum_api_user_get($uid);
                 $maildata = array();
+                $replace = array(
+                    "%phorum_title%" => $PHORUM['title'],
+                    "%login_url%" => phorum_get_url(PHORUM_LOGIN_URL),
+                );
+
                 $maildata['mailmessage'] =
-                   wordwrap($PHORUM["DATA"]["LANG"]["LostPassEmailBody1"],72).
+                   wordwrap(str_replace(array_keys($replace), array_values($replace), $PHORUM["DATA"]["LANG"]["LostPassEmailBody1"]),72).
                    "\n\n".
                    $PHORUM["DATA"]["LANG"]["Username"] .": $user[username]\n".
                    $PHORUM["DATA"]["LANG"]["Password"] .": $newpass".
                    "\n\n".
-                   wordwrap($PHORUM["DATA"]["LANG"]["LostPassEmailBody2"],72);
-                $maildata['mailsubject'] = $PHORUM["DATA"]["LANG"]["LostPassEmailSubject"];
+                   wordwrap(str_replace(array_keys($replace), array_values($replace), $PHORUM["DATA"]["LANG"]["LostPassEmailBody2"]),72);
+                $maildata['mailsubject'] = str_replace(array_keys($replace), array_values($replace), $PHORUM["DATA"]["LANG"]["LostPassEmailSubject"]);
                 phorum_email_user(array( 0 => $user['email'] ), $maildata);
 
                 $okmsg = $PHORUM["DATA"]["LANG"]["LostPassSent"];
